src/crate/testing/layer.py

```diff
@@ -269,6 +263,19 @@ def _wait_for(self, validator):
                 sys.stderr.write('.')
                 time.sleep(self.wait_interval)
 
+    def _wait_for_http_url(self, timeout=20):
```

Comment:
this was initially not a class method due to testability, however I just saw the method is not tested at all :(